dotCMS/src/main/java/com/dotcms/rest/ContentResource.java

```diff
@@ -463,6 +464,47 @@ public Response getContent(@Context HttpServletRequest request,
         return responseResource.response(result, null, status);
     }
 
+    /**
+     * This methods receives a Lucene query.
+     * It processes the query looking for special scenarios like structure fields (i.e: stInode, stName) and replace them with valid Content fields
+     * @param luceneQuery
+     * @return luceneQuery
+     */
+    private String processQuery(String luceneQuery) throws DotDataException, DotSecurityException {
+        if (luceneQuery == null) {
+            return null;
+        }
+
+        //Lucene parameters are separated by blankSpace
+        String [] params = luceneQuery.split(" ");
+        for (String paramStr : params) {
+
+            if (paramStr.contains(Contentlet.STRUCTURE_INODE_KEY + ":")) {
+                //Parameter is in the FORMAT  stInode:inode
+                String[] param = paramStr.split(":"); //Param and value pair separated by :
+                if (param.length == 2) {
+                    String inode = param[1];
+
+                    if (UtilMethods.isSet(inode)) {
+                        ContentType type = APILocator.getContentTypeAPI(APILocator.systemUser()).find(inode);
+                        if (type != null && InodeUtils.isSet(type.inode())) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)